test/WebJobs.Script.Tests.Shared/TestLoggerProvider.cs

```diff
@@ -20,7 +20,7 @@ public TestLoggerProvider(Func<string, LogLevel, bool> filter = null)
 
         private Dictionary<string, TestLogger> LoggerCache { get; } = new Dictionary<string, TestLogger>();
 
-        public IEnumerable<TestLogger> CreatedLoggers => LoggerCache.Values;
+        public IEnumerable<TestLogger> CreatedLoggers => LoggerCache.Values.ToList();
```

Comment:
Technically you can still get race conditions here, since the ToList does an enumeration. I wonder if you shouldn't just put  synchronization in place on the handful of accesses to LoggerCache in this file? Basically what we did for the TestLogger class  itself and it's _logMessages collection.

Reply:
you're right. I think I'll switch to a concurrent dictionary rather than than adding synchronization. Will update the PR.